hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -86,6 +83,20 @@ object HoodieWriterUtils {
     hoodieConfig.setDefaultValue(RECONCILE_SCHEMA)
     hoodieConfig.setDefaultValue(DROP_PARTITION_COLUMNS)
     hoodieConfig.setDefaultValue(KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED)
+    if (isFreshTable) { // only set default values for a fresh table. these might be used for config validation in subsequent commits and hence
+      // we should not setting defaults. (infer function could result in setting wrong defaults)
+      hoodieConfig.setDefaultValue(PAYLOAD_CLASS_NAME)
+      hoodieConfig.setDefaultValue(KEYGENERATOR_CLASS_NAME)
```

Comment:
What about payload ordering/precombine field, record key and partitionpath fields? They don't have any defaults, is it?

Reply:
I ran into some issues around precombine and hence left it out. Let me try to recollect.